### ISSUE TITLE
[Ubuntu] Weather details cannot be shown

### ISSUE THREAD

**Issue by ckorn** [dglent/meteo-qt#63]
Hi,

this is not really a bug but a feature request.
I want to use meteo on Ubuntu. I already created a package, installed it, got my key from openweathermap and added my keys. Now the icon shows the temperature but I cannot see the details because there is no tooltip in Ubuntu and also left click and right click have the same action (showing the drop down menu).

The whole thing is called appindicators:
https://unity.ubuntu.com/projects/appindicators/

Maybe you could add support for that.
Just a menu item to show the details in a dialog or something like that.

**Comment by dglent** [comment]
Hello,

So it is due this "feature":
"Application indicators are more consistent – no more left and right-click inconsistency. Always left click to see the items." I think that many applications will have a problem with this ...

I will add an entry in the right click menu to show the window, to see if it becomes usable with this system.

**Comment by ckorn** [comment]
Yeah, thanks to Canonical for breaking apps and thinking every dev out there cares about Ubuntu.

**Comment by dglent** [comment]
Here, in LXQT environment, this option works fine, could you test the above commit in Unity please ?

**Comment by ckorn** [comment]
Works!

Thanks for the quick fix !

![Image of Toggle Panel](http://storage6.static.itmages.com/i/16/1129/h_1480441914_9605082_2ca8ea03a2.png)

**Comment by dglent** [comment]
ok, nice!

**Comment by pmattern** [comment]
If I've got this right and the result from this request is that new topic "Toggle panel" at the top of the context menu as it can be seen e. g. on LXQt, then I doubt "panel" is a suitable term here.
While not plain wrong I figure it will be associated with the desktop environments' panels like e. g. lxqt-panel most of the time which in turn will give rise to confusion.
Couldn't this simply be called a "window"? As for German I took the liberty to translate it like that already, tbh.

That aside I think the tail is pretty much wagging the dog here.
Unity is taking a design decision that's pretty questionable. Associating the left click with the actual functionality and the right click with configuration is common and reasonable practice. Some (terrifying) "left and right-click inconsistency" should result in fixing the application in question and not in dropping one of the two kinds of clicks altogether.
The feature introduced in meteo-qt to mitigate this design decision of a single desktop environment on the other hand will sure be at best not helpful on a number of other desktop environments. IMO this doesn't make sense.
Could / should this topic be restricted to using meteo-qt in Unity?

**Comment by dglent** [comment]
Thanks for the suggestion, i will change the string to 'Toggle window', it is better.

You have 100% right for the left/right click.

It is truth that for other DE users, this Action in the right click menu is usefulness. 
It aims to workaround the Unity restriction, and allows people use the meteo-qt even in this DE. 

I found this discussion https://github.com/electron/electron/issues/6773 which describes very well the problem.
As we see, this is not considered as a bug from Ubuntu Unity, but as a feature request!
They don't care about the thousands of software which use the system tray icon.
They want to oblige devs to follow the users who follow them.
Until now it seems that even big projects couldn't "push" the things, and they have ended to adapt their programs to Unity.
Personally, this was one of the reason why i chose LXQt as my DE (I came from KDE).

Finally, i can check if it is possible to make the toggle window action appearing in the right menu only when running in the the Unity DE.
